src/Features/CSharp/Portable/EditAndContinue/SyntaxUtilities.cs

```diff
@@ -114,20 +114,29 @@ public static void AssertIsBody(SyntaxNode syntax, bool allowLambda)
             Debug.Assert(false);
         }
 
-        public static void FindLeafNodeAndPartner(SyntaxNode leftRoot, int leftPosition, SyntaxNode rightRoot, out SyntaxNode leftNode, out SyntaxNode rightNode)
+        public static void FindLeafNodeAndPartner(SyntaxNode leftRoot, int leftPosition, SyntaxNode rightRoot, out SyntaxNode leftNode, out SyntaxNode rightNodeOpt)
         {
             leftNode = leftRoot;
-            rightNode = rightRoot;
+            rightNodeOpt = rightRoot;
             while (true)
             {
-                Debug.Assert(leftNode.RawKind == rightNode.RawKind);
+                Debug.Assert(leftNode.RawKind == rightNodeOpt.RawKind);
                 var leftChild = leftNode.ChildThatContainsPosition(leftPosition, out var childIndex);
                 if (leftChild.IsToken)
                 {
                     return;
                 }
 
-                rightNode = rightNode.ChildNodesAndTokens()[childIndex].AsNode();
+                var rightNodeChildNodesAndTokens = rightNodeOpt.ChildNodesAndTokens();
```

Comment:
The VB implementation looks identical, so we'll probably want a corresponding fix and test in VB.